### ISSUE TITLE
Switch to using coffee-fmt package for beautifying CoffeeScript

### ISSUE THREAD

**Issue by shprink** [Glavin001/atom-beautify#192]
Hi,

Your plugin do not support CoffeeScript but still format it when saving (On save enabled).

it screws my doc block:

```
###
@ngdoc directive
@name wordpress-hybrid-client:wphcLoader
@restrict E
@description
A simple loader
@example
<pre>
</pre>
###
```

to this:

```
###
@ngdoc directive
@name wordpress - hybrid - client: wphcLoader
@restrict E
@description
A simple loader
@example
       < pre >
</pre >
###
```

When your plugin is disabled it is fine.

Thanks

**Comment by shprink** [comment]
Actually it seems that you support CoffeeScript but I do not see any specific options for it in the settings.

**Comment by shprink** [comment]
A really good feature would be to be able to activate/desactivate a specific file format. For instance this plugins https://github.com/atom/language-coffee-script seems to handle coffeescript better.

Cheers

**Comment by Glavin001** [comment]
CoffeeScript beautification support comes from https://github.com/derekchiang/Coffee-Formatter which is unfortunately not actively maintained. I am open to any alternatives.

> For instance this plugins https://github.com/atom/language-coffee-script seems to handle coffeescript better.

That supports CoffeeScript beautification?

**Comment by sterpe** [comment]
@Glavin001, @shprink I've been working on an npm module for coffee script reformatting, (like changing tabs to  arbitrary # of spaces or vice versa).  It's built on top of coffee-scripts own tokenizer (which is still pretty ambiguous).  Still very much beta, but it does at least transform the above docblock correctly.

Maybe you guys could check it out see how it performs.

https://www.npmjs.com/package/coffee-fmt

**Comment by Glavin001** [comment]
Great @sterpe ! I would love to support this and switch from Coffee-Formatter to your Coffee-fmt. Let me know once you think it's at an acceptable state for switching over and what the Node API is that I can use (see https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/coffeescript-beautify.coffee ). Thanks!

**Comment by Glavin001** [comment]
Hey @sterpe, what is the status of [coffee-fmt](https://github.com/sterpe/coffee-fmt)? If you are confident that it can replace Coffee-Formatter than I can support that. Let me know what you think :smiley:.

**Comment by Glavin001** [comment]
Published to v0.26.0

`coffee-fmt` is now the default CoffeeScript beautifier, deprecating Coffee-Formatter. Thanks for the awesome work so far, @sterpe !
